src/main/java/broker/Log.java

```diff
@@ -0,0 +1,72 @@
+package broker;
+
+import java.util.List;
+
+/**
+ * Log represents the entire log for a single partition in Flux, and more specifically is composed of
+ * multiple LogSegment objects in which each segment represents a chunk of the partition's data on disk.
+ * Log, by extension, is essentially a manager for all these segments.
+ */
+public class Log {
+    private List<LogSegment> segments;
+    private int currentActiveSegmentIdx = 0; // references index in above list
+    private int logStartOffset; // record offsets, not bytes
+    private int logEndOffset;
+    private long currentSizeInBytes;
+
+    // By default, creating a Log will also instantiate 1 empty and mutable LogSegment (default record offset: 0)
+    public Log() {
+        this.segments = List.of(new LogSegment(0, 0));
```

Comment:
i noticed that this class has a method for adding segments. `List.of` will return an immutable list so an error will be thrown if someone tries to add to the list

Reply:
^ This is non-blocking, more of a heads up that `List.add` won't work for adding new items

Reply:
O wait, just re-read the line, it looks like the LogSement themselves aren't being appended, but instead, stuff is being appended to the LogSegment. You can ignore my comment above, but the point still stands that if the lists that are being appended to use `List.of`, then `List.add` won't work

Reply:
good catch, forgot `List.of()` returns an immutable list. the segments list in the class should be mutable since more segments will be created as "old" ones reach a size threshold